website/src/components/APITable/index.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {
+  Children,
+  ComponentProps,
+  ReactElement,
+  ReactNode,
+  isValidElement,
+  useRef,
+  useEffect,
+} from 'react';
+import useIsBrowser from '@docusaurus/useIsBrowser';
+import styles from './styles.module.css';
+
+interface Props {
+  readonly children: ReactElement<ComponentProps<'table'>>;
+  readonly name?: string;
+}
+
+// ReactNode equivalent of HTMLElement#innerText
+function getText(node: ReactElement): string {
+  let curNode: ReactNode = node;
+  while (isValidElement(curNode)) {
+    [curNode] = Children.toArray(curNode.props.children);
+  }
+  return curNode as string;
+}
+
+/*
+ * Note: this is not a quite robust component since it makes a lot of
+ * assumptions about how the children looks; however, those assumptions
+ * should be generally correct in the MDX context.
+ */
+export default function APITable({children, name}: Props): JSX.Element {
+  const isBrowser = useIsBrowser();
+  const [thead, tbody] = Children.toArray(
+    children.props.children,
+  ) as ReactElement[];
+  const highlightedRow = useRef<HTMLTableRowElement>(null);
+  const rows = Children.map(
+    tbody.props.children,
+    (row: ReactElement<ComponentProps<'tr'>>) => {
```

Comment:
Can you extract an `<APITableRow>` component, and encapsulate the highlighting logic inside it?

Also use of `useHistory` rather than DOM APIs that do not trigger re-renders on route changes (in this case it might not produce any weird side effects but is still a bad practice to access window during render execution)